### ISSUE TITLE
rspamd processes crash

### ISSUE THREAD

**Issue by chri2** [YunoHost/issues#2269]
### Describe the bug

rspamd crashes regularly:

```2023-10-17 13:27:42 #1368(main) <316489>; main; rspamd_check_termination_clause: normal process 26132 terminated abnormally with exit code 0 by signal: Speicherzugriffsfehler but NOT created core file (throttled=no); core file limits: 0 current, -1 max
2023-10-17 13:27:42 #26134(normal) rspamd_crash_sig_handler: caught fatal signal 11(Speicherzugriffsfehler), pid: 26134, trace: 
2023-10-17 13:27:42 #1368(main) <316489>; main; rspamd_check_termination_clause: normal process 26134 terminated abnormally with exit code 0 by signal: Speicherzugriffsfehler but NOT created core file (throttled=no); core file limits: 0 current, -1 max
2023-10-17 13:27:43 #26140(normal) rspamd_crash_sig_handler: caught fatal signal 11(Speicherzugriffsfehler), pid: 26140, trace: 
2023-10-17 13:27:43 #1368(main) <316489>; main; rspamd_check_termination_clause: normal process 26140 terminated abnormally with exit code 0 by signal: Speicherzugriffsfehler but NOT created core file (throttled=no); core file limits: 0 current, -1 max
2023-10-17 13:27:45 #28514(normal) rspamd_crash_sig_handler: caught fatal signal 11(Speicherzugriffsfehler), pid: 28514, trace: 
2023-10-17 13:27:45 #1368(main) <316489>; main; rspamd_check_termination_clause: normal process 28514 terminated abnormally with exit code 0 by signal: Speicherzugriffsfehler but NOT created core file (throttled=no); core file limits: 0 current, -1 max
```

### Context

- Hardware: RockPro64 + 2x SATA SSD as software raid level 1
- YunoHost version: 11.2.5
- I have access to my server: Through SSH | through the webadmin
- Are you in a special context or did you perform some particular tweaking on your YunoHost instance?: *yes*
  - If yes, please explain: **multiple domains for email, added aliases as described [here](https://forum.yunohost.org/t/add-a-redirection-mail-alias-without-attaching-it-to-a-yunohost-account/182)**

### To reproduce

* Look at your logfile. No?
* Look at the logfile of your Rock64Pro yunohost. No?
* Look at the logfile of your Rock64Pro SATA SSDs software raid 1 yunohost. No?

Can't say so far. I can increase the rate of the error significantly by running `rspamc learn_spam /var/mail/<user>/.junk/cur` on a big junk mailbox.

### Expected behavior

No crashes at all.

### Logs

On systems affected 

```
for P in $(grep Speicherzugriffsfehler /var/log/rspamd/rspamd.log | perl -pe's/.*#(\d+)\(.*/$1/' | sort | uniq); do grep "#$P(" /var/log/rspamd/rspamd.log; done
```
should show the logs of the processes that crashed.

**Comment by chri2** [comment]
I tried to dig a bit more and switched my system wide locale from `de_DE.UTF-8` to `C.UTF-8` to get the rspamd logfile in pure english without german mixed in. Restarted rspamd using `systemctl`.

Not reproducible.

Switched the locale back to `de_DE.UTF-8`, restarted rspamd and reproduced the bahviour.

To get my ynh working with rspamd I switched back to  `C.UTF-8`, restarted rspamd and got the same errors containing german again. WTF.

After I restarted the daemon a few times and tried after each time to learn more ham/spam from existing folders it suddenly started working without crashes again.

I guess someone needs to look at the problem on their host to compare results.
